src/main/java/com/reservation/instructor/domain/Instructor.java

```diff
@@ -0,0 +1,65 @@
+package com.reservation.instructor.domain;
+
+import com.reservation.instructor.domain.exception.InstructorAlreadyActiveException;
+import com.reservation.instructor.domain.exception.InstructorAlreadyInactiveException;
+import com.reservation.instructor.domain.exception.ProfileCannotBeUpdatedWhenInactiveException;
+import lombok.EqualsAndHashCode;
+
+import java.util.Objects;
+
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+public class Instructor {
+    @EqualsAndHashCode.Include
+    private final InstructorId instructorId;
+    private final SystemUserId systemUserId;
+    private InstructorProfile profile;
+    private InstructorStatus status;
+
+    private Instructor(InstructorId instructorId, SystemUserId systemUserId, InstructorProfile profile) {
+        this.instructorId = Objects.requireNonNull(instructorId);
+        this.systemUserId = Objects.requireNonNull(systemUserId);
+        this.profile = Objects.requireNonNull(profile);
+        this.status = InstructorStatus.ACTIVE;
+    }
+
+    public static Instructor create(InstructorId instructorId, SystemUserId systemUserId, InstructorProfile profile) {
+        return new Instructor(instructorId, systemUserId, profile);
+    }
+
+    public InstructorId getInstructorId() {
+        return this.instructorId;
+    }
+
+    public SystemUserId getSystemUserId() {
+        return this.systemUserId;
+    }
+
+    public void updateProfile(InstructorProfile profile) {
+        if(!isActive()) {
+            throw new ProfileCannotBeUpdatedWhenInactiveException();
+        }
+        this.profile = Objects.requireNonNull(profile);
+    }
```

Comment:
Myślę, że warto dodać obiekt, który będzie reprezentował zdarzenie domenowe, dokładnie to co dzieje się w poprawkach w PR rsb-89